webhook.js

```diff
@@ -0,0 +1,62 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const morgan = require('morgan');
+const crypto = require('crypto');
+const bufferEq = require('buffer-equal-constant-time')
+const app = express();
+const port = 8000;
+const tolerance = 0.00001;
+const secretKey = "Your secret key stored in hook.config.secret"  // never store this in code!
+
+app.use(morgan('combined'));
+app.use(bodyParser.json({verify:function(request,res,buf){request.rawBody=buf}}))
+
+function verifySignature(request) {
+  const retrievedSignature = request.get("X-Elis-Signature").split("=")
+  const bodyString = Buffer.from(request.rawBody)
+  if (retrievedSignature[0] !== "sha1") {
+    return false
+  }
+  const hmac = crypto.createHmac('sha1', secretKey).update(bodyString).digest('hex')
+  return bufferEq(new Buffer.from(hmac), new Buffer.from(retrievedSignature[1]));
+};
+
+const findByPath = (data, [part, ...parts]) => {
+  const found = data.find(({ schema_id }) => (part === schema_id));
+  return (found.children && parts.length) ? findByPath(found.children, parts) : found;
+};
+
+const createErrorMessages = (id, content) => [{ id, type: 'error', content }];
+
+const trycatch = (fn) => (data) => {
+  try { return fn(data); } catch (e) { return createErrorMessages('all', 'Error parsing annotation'); }
+};
+
+const floatValue = ({ value }) => parseFloat(value);
```

Comment:
OK! ideally, `floatValue` function should look like this: `({ content: { value } }) => parseFloat(value)` and you can ignore my previous comments 😄